app/controllers/users_controller.rb

```diff
@@ -17,25 +17,14 @@ def new_invited
 
   def create
     @user = User.new(user_params)
-    if @user.valid?
-      Stripe.api_key = ENV["STRIPE_SECRET_KEY"]
-      charge = StripeWrapper::Charge.create(
-        :amount => 999,
-        :source => params[:stripeToken],
-        :description => "MyFlix Sign Up Charge for #{@user.email}"
-      )
-      if charge.successful?
-        @user.save
-        handle_invitation
-        AppMailer.welcome_email(@user).deliver
-        flash[:success] = "You have successfully registered. You can sign in now!"
-        redirect_to sign_in_path
-      else
-        flash.now[:danger] = charge.error_message
-        render :new
-      end
+    result = UserSignup.perform(user: @user,
+                                invitation_token: params[:invitation_token],
+                                stripe_token: params[:stripeToken])
```

Comment:
Agreed @tomtomecek. May be because I am used to seeing it now is why I find it easier to read. I did take me a couple of reads to get my head around it :smile: